src/shorah/shorah_snv.py

```diff
@@ -400,23 +403,82 @@ def main(args):
+        VCF_meta = [
+            '##fileformat=VCFv4.2',
+            '##fileDate={:%Y%m%d}'.format(date.today()),
+            '##source=ShoRAH_{}'.format(args.version),
+            '##reference={}'.format(args.f),
+            '##INFO=<ID=Frq<X>,Number=1,Type=Float,Description="Frequency of the variant in window <X>">',
+            '##INFO=<ID=Post<X>,Number=1,Type=Float,Description="Posterior probability of the variant in window <X>">',
+            '##INFO=<ID=Fvar,Number=1,Type=Integer,Description="Number of forward reads with variant">',
+            '##INFO=<ID=Rvar,Number=1,Type=Integer,Description="Number of reverse reads with variant">',
+            '##INFO=<ID=Ftot,Number=1,Type=Integer,Description="Total number of forward reads">',
+            '##INFO=<ID=Rtot,Number=1,Type=Integer,Description="total number of reverse reads">',
+            '##INFO=<ID=Pval,Number=1,Type=Float,Description="P-value for strand bias">',
+            '##INFO=<ID=Qval,Number=1,Type=Float,Description="Q-value for strand bias">',
+        ]
+
+        with open(VCF_file, 'w') as vcf:
+            vcf.write('\n'.join(VCF_meta))
+            # VCFv4.2 HEADER line
+            vcf.write('\n#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO')
+            # Iterate over single SNV lines and write them to output file
+            for wl in write_list:
+                # only print when q >= 5%
+                if wl[-1] >= 0.05:
+                    info = 'Fvar={};Rvar={};Ftot={};Rtot={};Pval={};Qval={}' \
+                        .format(*wl[-6:])
+                    if increment == 1:
+                        post_avg = min([1, float(wl[5])])
+                        info = 'Freq={};Post={};'.format(*wl[4:6]) + info
+                    else:
+                        info = 'Freq1={};Freq2={};Freq3={};Post1={};Post2={};Post3={};' \
+                                .format(*wl[4:10]) \
+                            + info
+                        post_all = []
+                        for freq, post in zip(wl[4:7], wl[7:10]):
+                            if freq == '*':
+                                pass
+                            elif freq == '-':
+                                post_all.append(0)
+                            else:
+                                post_all.append(min([1, float(post)]))
+                        # Calculate posterior average
+                        post_avg = sum(post_all) / len(post_all)
+                    # Calcualte a Phred quality score where the base calling 
+                    # error probabilities is set to (1 - posterior avg). 
+                    # Maximum is set to 100.
+                    try:
+                        qual_norm = -10 * log10(1 - post_avg)
+                    except ValueError:
+                        qual_norm = 100
+                    snv = wl[:4] + [qual_norm, info]
+                    vcf.write('\n{}\t{}\t.\t{}\t{}\t{}\tPASS\t{}'.format(*snv))
```

Comment:
Just came to my mind. We should probably use the ALT field to indicate whether a deletion is found, instead of writing always a dot

Reply:
'.' is always written to the ID column. In case of a deletion, the ALT column contains a '-'